src/compiler/crystal/command.cr

```diff
@@ -309,7 +307,10 @@ class Crystal::Command
       end
 
       unless no_codegen
-        opts.on("--emit [#{VALID_EMIT_VALUES.join('|')}]", "Comma separated list of types of output for the compiler to emit") do |emit_values|
+        valid_emit_values = {{ Compiler::EmitKinds.constants.map(&.stringify).reject { |m| %w(none all).includes?(m.downcase) } }}
```

Comment:
Why use a macro instead of `Enum.names`?

Reply:
Wow how did I miss that (PR-ing at 3am may be why :smile: )
thanks!